### ISSUE TITLE
Running powershell script in pom.xml goals fails on linux only agents

### ISSUE THREAD

**Issue by cotovanu-cristian** [jenkins-infra/jenkins-security-scan#34]
### Reproduction steps

1. In the `pom.xml` run a powershell script 
e.g: 
```
<execution>
    <id>Generate help files</id>
    <phase>generate-resources</phase>
    <goals>
        <goal>run</goal>
    </goals>
    <configuration>
        <target>
            <exec executable="powershell.exe">
                <arg value=".\GenerateHelpFiles.ps1 " />
            </exec>
        </target>
    </configuration>
</execution>
```
2. Run the Jenkins Security Scan on PR

### Expected Results

Expected the Scan to execute

### Actual Results

The scan failed cause the agent did not have `powershell.exe`

![chrome_ExFVAp8QjC](https://github.com/user-attachments/assets/a7227e9a-24c8-4287-931f-3200e67ab175)

### Anything else?

In the `pom.xml` we run a powershell script running in the build goals, but the agent building the plugin in the security scan script is a linux agent thus it won't be capable of building the plugin used for the security assesment (https://github.com/jenkinsci/uipath-automation-package-plugin/blob/develop/pom.xml#L120C1-L126C41).
What can we do to onboard to the `Jenkins Security Scan`?

**Comment by cotovanu-cristian** [comment]
scan example: https://github.com/jenkinsci/uipath-automation-package-plugin/actions/runs/11416380208/job/31767532043

**Comment by cotovanu-cristian** [comment]
I managed to fix this by using `pwsh` instead of `powershell.exe`. ✅ 
💡 But adding the option to build on windows agents would be a nice addition. 

e.g:
```
<execution>
    <id>Generate help files</id>
    <phase>generate-resources</phase>
    <goals>
        <goal>run</goal>
    </goals>
    <configuration>
        <target>
            <exec executable="pwsh">
                <arg value=".\GenerateHelpFiles.ps1 " />
            </exec>
        </target>
    </configuration>
</execution>
```

**Comment by daniel-beck** [comment]
Thanks for reporting this. I admit I did not consider builds that require Windows to pass when I created this, as that's not common in the Jenkins project. GitHub offers Windows runners, so this should be doable with a few changes. Fixing this is not a very high priority for me at the moment, as this is a rare use case.

Meanwhile I recommend looking into using the [published package](https://github.com/jenkins-infra/jenkins-codeql/pkgs/container/jenkins-codeql) from a standard CodeQL workflow, for which Jenkins Security Scan is currently just a fairly small wrapper. Looking at the [default CodeQL workflow](https://github.com/actions/starter-workflows/blob/main/code-scanning/codeql.yml) it doesn't look like it is intended to run with Windows based on labels, but perhaps it's OS agnostic enough to work -- I see a few references to Windows in https://github.com/github/codeql-action, so might be worth a try. See [the package's readme](https://github.com/jenkins-infra/jenkins-codeql?tab=readme-ov-file#use-in-a-regular-codeql-workflow) for instructions for use in a generic CodeQL workflow.

(Edited to add: I wrote this hours ago, before your latest comment, and only submitted now 😑 Leaving it in case it's useful for someone who _really_ needs Windows for the build.)
